src/wire/ipv6ext.rs

```diff
@@ -0,0 +1,246 @@
+use {Error, Result};
+
+enum_with_unknown! {
+    /// IPv6 Extension Header Option type
+    pub doc enum OptionType(u8) {
+        /// Pad1 option
+        Pad1 =  0,
+        /// PadN option
+        PadN =  1
+    }
+}
+
+/// A read/write wrapper around a variable number of IPv6 Extension Header Options
+#[derive(Debug, PartialEq)]
+pub struct Packet<T: AsRef<[u8]>> {
+    buffer: T
+}
+
+// Format of Option
+//
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+- - - - - - - - -
+// |  Option Type  |  Opt Data Len |  Option Data
+// +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+- - - - - - - - -
+//
+//
+// See https://tools.ietf.org/html/rfc8200#section-4.2 for details.
+mod field {
+    use wire::field::*;
+
+    // 8-bit identifier of the type of option
+    pub const TYPE:     usize = 0;
+    // 8-bit unsigned integer. Length of the DATA field of this option, in octets
+    pub const LENGTH:   usize = 1;
+	// Variable-length field. Option-Type-specific data.
+    pub const DATA:     Rest  = 2..;
+}
+
+impl<T: AsRef<[u8]>> Packet<T> {
+    /// Create a raw octet buffer with an IPv6 Extension Header Option packet structure.
+    pub fn new(buffer: T) -> Packet<T> {
```

Comment:
A `check_len` function should be added, and a `new_checked` function should be added. See some of the other implementations in `wire` for details. Make sure to test `check_len` well. It will be a little tricky since the length requirements vary based on the options type.

Reply:
Added `check_len` and `new_checked`. I am not sure that I handled `Pad1 correctly. As is, calling `option_data_length` or `data` methods will panic if the type is `Pad1`.